backend/fms_core/viewsets.py

```diff
@@ -422,8 +443,74 @@ class SampleKindViewSet(viewsets.ModelViewSet):
     queryset = SampleKind.objects.all()
     serializer_class = SampleKindSerializer
     pagination_class = None
-    permission_classes = [AllowAny]
+    permission_classes = [IsAuthenticated]
+
+class ProtocolViewSet(viewsets.ModelViewSet):
+    queryset = Protocol.objects.all()
+    serializer_class = ProtocolSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated]
+
+class ProcessSampleViewSet(viewsets.ModelViewSet, TemplateActionsMixin):
+    queryset = ProcessSample.objects.all().select_related("process").prefetch_related("lineage")
+    queryset = queryset.annotate(child_sample=F("lineage__child"))
+    queryset = queryset.annotate(child_sample_name=F("lineage__child__name"))
+    queryset = queryset.annotate(source_sample_name=F("source_sample__name"))
+
+    serializer_class = ProcessSampleSerializer
+
+    ordering_fields = (
+        *_list_keys(_process_sample_filterset_fields),
+    )
+
+    filterset_fields = {
+        **_process_sample_filterset_fields,
+    }
+
+    template_action_list = [
+        {
+            "name": "Process Extractions",
+            "description": "Upload the provided template with up to 96 extractions.",
+            "template": SAMPLE_EXTRACTION_TEMPLATE,
+            "resource": ExtractionResource,
+        },
+    ]
+
+    @action(detail=False, methods=["get"])
+    def list_export(self, _request):
+        serializer = ProcessSampleExportSerializer(self.filter_queryset(self.get_queryset()), many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["get"])
+    def search(self, _request):
+        """
+        Searches for process sample that match the given query
+        """
+        search_input = _request.GET.get("q")
+
+        query = Q(id__icontains=search_input)
 
+        process_sample_data = ProcessSample.objects.filter(query)
+        page = self.paginate_queryset(process_sample_data)
+        serializer = self.get_serializer(page, many=True)
+        return self.get_paginated_response(serializer.data)
+
+    @action(detail=False, methods=["get"])
+    def summary(self, _request):
+        """
+        Returns summary statistics about the current set of ProcessSample in the
+        database.
+        """
+
+        protocol_names_by_id = {protocol.id: protocol.name for protocol in Protocol.objects.all()}
+
+        return Response({
+            "total_count": ProcessSample.objects.all().count(),
+            "protocol_counts": {
+                protocol_names_by_id[c["process__protocol"]]: c["process__protocol__count"]
+                for c in self.queryset.values("process__protocol").annotate(Count("process__protocol"))
+            },
+        })
```

Comment:
Do we really want to have process by name rather than by ID in the frontend? What's the advantage of having them by name? I'd rather have a single identifier.

Reply:
I was using the names for protocol to copy the way this is done with the sample kind but it does not have to be name. We could use the id's instead, but having a human readable summary may be a good thing to have... I am not sure if that counts for anything in our case though.

Reply:
ok changed the summaries to use IDs for protocols and sample kinds.